rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -144,24 +167,22 @@ def __next__(self):
             self.labels_padded = torch.LongTensor([row + [0] * (max_cols1 - len(row)) for batch in self.labels_padded for row in batch])
             self.labels_padded = self.labels_padded.view(-1, max_rows1, max_cols1)
 
-            if self.tensor_dtype == types.FLOAT:
-                return torch.from_numpy(self.out),self.bb_padded, self.labels_padded
-            elif self.tensor_dtype == types.FLOAT16:
-                return torch.from_numpy(self.out.astype(np.float16)),self.bb_padded, self.labels_padded
+            return self.out,self.bb_padded, self.labels_padded
 
         else:
             if(self.loader._oneHotEncoding == True):
                 self.loader.GetOneHotEncodedLabels(self.labels)
```

Comment:
please pass device type (host or gpu) for GetOneHotEncodedLabels, so that it can copy to the correct destination. In case of HIP backend, please copy from internal rocal cpu memory to HIP mem pointer of the labels tensor, so we can avoid all .cuda() calls in the training script

Reply:
Done